app/models/concerns/foreman_remote_execution/orchestration/ssh.rb

```diff
@@ -0,0 +1,58 @@
+module ForemanRemoteExecution
+  module Orchestration::SSH
+    extend ActiveSupport::Concern
+
+    included do
+      before_destroy :ssh_destroy
+      after_validation :queue_ssh_destroy
+      register_rebuild(:queue_ssh_destroy, N_("SSH_#{self.to_s.split('::').first}"))
+    end
+
+    def drop_from_known_hosts(args)
+      proxy_id, target = args
+      proxy = ::SmartProxy.find(proxy_id)
+      begin
+        proxy.drop_host_from_known_hosts(target)
+      rescue => e
+        Rails.logger.warn e.message
+        return false
+      end
+      true
+    end
+
+    def ssh_destroy
+      logger.debug "Scheduling SSH known_hosts cleanup"
+
+      host, _kind, target = host_kind_target
+      proxies = host.remote_execution_proxies('SSH').values
+      proxies.flatten.uniq.each do |proxy|
```

Comment:
no need to flatten now, rith?
```suggestion
      host.remote_execution_proxies('SSH').values.each do |proxy|
```

Reply:
We cannot, `host.remote_execution_proxies('SSH')` returns a hash of `Symbol -> Array<SmartProxy>`, so at this point, we're operating on `Array<Array<SmartProxy>>` and we need to flatten (and possibly even `uniq`) it

Reply:
ah, got it, thaks for expelation :+1: